### PR TITLE
#16 Sidebar Journal Listing HTML

### LINKED ISSUE #15: [JAVASCRIPT] Integrate "New Journal" Button to Allow New Journal Listing
As a user I want to be able to write my own title and create a new journal so that I can write different stuff

Spec:
- Clicking on the "New Journal" button should display new journals listed below

### LINKED ISSUE #16: [FRONTEND] Sidebar Journal Listing HTML
As a user I want to be able to see all the journals that I created such that I can navigate to them with ease.

Spec:

- Journals are clearly listed on the left side bar.
- Styling inherits necessary global CSS variables.
- 3 '...' button on the side for quick actions (should display on hover)
- Component should be able to be updated dynamically.

### PR DESCRIPTION
**Pull request by Summerr12**
Issue #16

### CODE REVIEW

**Comment by reisandylamdjani** [comment]
can you change the commit message to be [feat] (your commit message)

File: application/web/index.html [hunk 1]
```diff
@@ -7,6 +7,7 @@
     <meta name="description" content="Instone; an effective markdown editor.">
     <title>Instone</title>
     <link rel="stylesheet" href="main.css">
+    <script src="web.js"></script>
```
Comment (camman00) [inline application/web/index.html]: Nice!

File: application/web/index.html [hunk 2]
```diff
@@ -24,6 +25,10 @@
                     <span>My Journal</span>
                     <i class='fa fa-ellipsis-v'></i>
                 </div>
+                <div class="journal-entry">
+                    <span>My Journal</span>
+                    <i class='fa fa-ellipsis-v'></i>
```
Comment (camman00) [inline application/web/index.html]: This has now been replaced with vdots img class.

File: application/web/main.css [hunk 3]
```diff
@@ -62,6 +62,18 @@ sidebar {
...
+    /* to make journal list for too many journals */
+    overflow-y: scroll;
+}
+
+.sidebar-module::-webkit-scrollbar {
+    width: 0px; 
+}
+
+
+.sidebar-module .fa.fa-ellipsis-v{
```
Comment (camman00) [inline application/web/main.css]: This will need to change to .vdots

File: application/web/main.css [hunk 4]
```diff
@@ -103,6 +115,7 @@ button.new-journal img.journal-icon {
     align-items: center;
     cursor: pointer;
     font-size: 16px;
+    margin-bottom: 10px;
```
Comment (camman00) [inline application/web/main.css:155]: I committed a breaking change for this potentially.

File: application/web/web.js [hunk 5]
```diff
@@ -1,3 +1,22 @@
...
+
+        const faIcon = entry.querySelector('.fa.fa-ellipsis-v');
+
+        entry.addEventListener('mouseover', () => {
+            faIcon.style.display = 'block';
+        });
+
+        entry.addEventListener('mouseout', () => {
+            faIcon.style.display = 'none';
+        });
```
Comment (camman00) [inline application/web/web.js]: Just a suggestion :)

Might be worth looking up 'event delegation'

Here is a code snippet: 

`document.querySelector('#parent-element').addEventListener('click', function(event) {
  const targetElement = event.target.closest('.child-element');
  if (targetElement) {
    // Handle the event for the child element
    console.log('Child element clicked:', targetElement);
  }
});`

**Review by camman00 — CHANGES_REQUESTED** [review]
Requesting changes. Please git pull from main since breaking changes were pushed (which might resolve some issues automatically 👍 ) Let us know if you have any questions!

File: application/web/main.css (hunk 4 above)
Comment (Summerr12) [reply application/web/main.css:155]: When I pulled I saw the line 'min-height: 40px;' but I dont notice the change that puts space between the my journals
image

**Comment by Summerr12** [comment]
Applied event delegation and tested on new journal entries, inturn making JS for issue 15#

**Comment by Summerr12** [comment]
Made the button create new entries which was part of issue 15. I kept it here since when I tried to make a new branch, the margins werent edited yet.
image

File: application/web/web.js [hunk 6]
```diff
@@ -10,5 +13,44 @@ window.onload = () => {
         
         const journalEntryBtn = document.querySelector('.journal-entry');
         journalEntryBtn.classList.toggle('toggled-journal-entry');
-    }
+    };
+};
+
+function createJournalEntries(){
+    const journal = document.querySelector('.new-journal');
+    journal.addEventListener('click', (event) => {
```
Comment (reisandylamdjani) [inline application/web/web.js]: I think you can remove the event argument

**Review by reisandylamdjani — CHANGES_REQUESTED** [review]

**Comment by reisandylamdjani** [comment]
there's a new merge to main, so you might want to pull again and merge to your branch, and merge conflict in web.js

**Comment by reisandylamdjani** [comment]
Ben could you remove line 27-32 in index.html so that when the initial page is loaded, we don't see the grey "My Journal" stuff

**Review by reisandylamdjani — APPROVED** [review]
Looks good now!

**Pull request merged**